clang/include/clang/AST/OpenMPClause.h

```diff
@@ -342,6 +342,57 @@ template <class T> class OMPVarListClause : public OMPClause {
   }
 };
 
+template <class T> class OMPDirectiveListClause : public OMPClause {
```

Comment:
Why do you need this?

Reply:
The idea was to factor out common functionality shared by absent/contains clauses. It doesn't do a huge amount, admittedly, but I think it's still better than copy/pasting a bunch of code for the absent/contains clause classes. (OMPNoChildClause, OMPOneStmtClause are similar in intent though perhaps wider in applicability).

It should probably have a comment, if that's what you meant!